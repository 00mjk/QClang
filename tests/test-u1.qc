qubit u1(float lambda, qubit q) {
    return U(0., 0., lambda, q);
}

qubit u2(float phi, float lambda, qubit q) {
    return U(3.14159265 / 2., phi, lambda, q);
}

int main() {
    qubit q1;
    qubit q2;

		q1 = false;
		q2 = false;
    u1(1.618, q1);
    u2(3.268, 6.28, q2);
    return 0;
}
