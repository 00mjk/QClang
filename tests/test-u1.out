OPENQASM 2.0;
include "qelib1.inc";
qreg q[1];
creg c[1];
h q;
qreg temp_0[1];
qreg temp_1[1];
U(0., 0., 1.618) temp_0;
U(1.570796325, 3.268, 6.28) temp_1;
